### PR TITLE
Contact Us form

### PR DESCRIPTION
**Pull request by PeterBreen**
Updates the Contact Us page to include an HTML contact form, using `react-google-recaptcha` to provide google reCaptcha for antispam.

The form will send an email on behalf of a user and allow us to control where it is sent, then reply directly to the person asking the question.

The PR also has a guard function against double or triple-submitting based on multiple inputs like an accidental double click on the submit button. It can be used like so:`Guard.duplicateInput(function, duration, options)` and is a wrapper for lodash debounce at the moment, so duration and options should be passed per [lodash's docs](https://lodash.com/docs/4.17.15#debounce). By default, duration = 1000, options = {leading: true}.

### CODE REVIEW

**Comment by PeterBreen** [comment]
This PR requires setting three new env keys:
export GOOGLE_RECAPTCHA_SITE_KEY = our site key (can be public)
export GOOGLE_RECAPTCHA_SECRET_KEY = our google secret key (obviously keep this secret)
export CONTACT_EMAIL = email address we want the contact form to target

File: democracylab/emails.py [hunk 1]
```diff
@@ -292,3 +292,14 @@ def _get_co_owner_emails(project):
 
 def _get_account_from_email(email_acct):
     return email_acct['from_name'] if email_acct is not None else 'DemocracyLab'
+
+def contact_democracylab_email(fname, lname, emailaddr, body):
+    subject = '{} {} would like to contact DemocracyLab'.format(fname, lname)
+    email_msg = EmailMessage(
+        subject=subject,
+        body=body,
+        from_email=[emailaddr],
```
Comment (marlonkeating) [inline democracylab/emails.py]:
```suggestion
        from_email=_get_account_from_email(settings.EMAIL_SUPPORT_ACCT),
```
from_email is an account object, it's not enough to specify just an email address.

File: democracylab/settings.py [hunk 2]
```diff
@@ -160,6 +160,7 @@ def read_connection_config(config):
 
 PROTOCOL_DOMAIN = os.environ['PROTOCOL_DOMAIN']
 ADMIN_EMAIL = os.environ['ADMIN_EMAIL']
+CONTACT_EMAIL = os.environ['CONTACT_EMAIL']
```
[the remark this answers is not in the document]
Comment (PeterBreen) [reply democracylab/settings.py]: Good catch, thank you. Added that.

File: common/components/forms/ContactForm.jsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+      lname: '',
+      emailaddr: '',
+      message: '',
+      sendStatusMessage: null,
+      sendStatusClass: null,
+      reCaptchaValue: null,
+    };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = _.debounce(this.handleSubmit.bind(this),1000, {'leading': true});
```
[the remark this answers is not in the document]
Comment (PeterBreen) [reply common/components/forms/ContactForm.jsx]: If it's the best approach for guarding against multiple submits (or the approach we just decide on) then, yeah, makes sense. Unless there's another idea that might work better?

[the remark this answers is not in the document]
Comment (marlonkeating) [reply common/components/forms/ContactForm.jsx]: Even if we pick a different multi-click guard implementation later on, it makes sense to put it in a common helper method so we only have to change it in one place later.

File: common/components/forms/ContactForm.jsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
+// @flow
+
+import React from 'react';
+import ProjectAPIUtils from '../utils/ProjectAPIUtils.js';
+import ReCAPTCHA from "react-google-recaptcha";
+import _ from 'lodash';
+
+class ContactForm extends React.Component {
```
[the remark this answers is not in the document]
Comment (PeterBreen) [reply common/components/forms/ContactForm.jsx]: I agree with the last three but I've got reservations about the first. Nested state has some real pitfalls for performance and code cleanliness; see [this Stackoverflow thread](https://stackoverflow.com/a/51136076) for example. Generally I'm 100% on board with you that we need to be consistent in pattern but this one I'd hesitate on.

[the remark this answers is not in the document]
Comment (marlonkeating) [reply common/components/forms/ContactForm.jsx]: Okay, you can keep the form fields directly in state.  A couple tips:
- If you want to separate out the form fields in flow to keep things straight, you can concatenate two types together and do something like type State = FormFields & OtherStateVariablesThatAreNotFormFields
- If you don't use a formFields state variable, make sure to pass in a custom getter and setter into form.setup().

File: common/components/utils/guard.js [hunk 5]
```diff
@@ -3,7 +3,7 @@
 import _ from 'lodash';
 
 class Guard {
-  static click(func: string, duration: number, options: object): string {
+  static click(func: string, duration = 1000: number, options = {'leading': true}: object): string {
```
Comment (PeterBreen) [inline common/components/utils/guard.js]: I added these if we want to just write `Guard.click(function)` in a given component rather than specifying out duration and options each time. I think these are reasonable enough default values for our purposes.

Comment (marlonkeating) [reply common/components/utils/guard.js]: Sounds reasonable.  My only comment is that this function as laid out could be used for things besides clicking, so we might want to rename it to something more generic.

Comment (PeterBreen) [reply common/components/utils/guard.js]: Renamed it to Guard.duplicateInput - so it reads as "guard (against) duplicate inputs," which is the most generic name for it I can think of. If something better comes to mind it's an easy fix right now.

File: common/components/forms/ContactForm.jsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+      lname: '',
+      emailaddr: '',
+      message: '',
+      sendStatusMessage: null,
+      sendStatusClass: null,
+      reCaptchaValue: null,
+    };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = Guard.duplicateInput(this.handleSubmit.bind(this), 1000, {'leading': true});
```
Comment (marlonkeating) [inline common/components/forms/ContactForm.jsx]:
```suggestion
    this.handleSubmit = Guard.duplicateInput(this.handleSubmit.bind(this));
```
No need to provide arguments when they match the default.

Comment (PeterBreen) [reply common/components/forms/ContactForm.jsx]: Makes sense, updated.

File: common/components/forms/ContactForm.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (PeterBreen) [reply common/components/forms/ContactForm.jsx]: I took some time to step back and think about this while researching some of the errors I'm getting, and I am no longer convinced we should be refactoring this page to use the Create Project form model. 

First, code reuse: I do not feel I am reusing enough of the Create Project form code. Custom form getter and setter, different view, different needs on the backend (submit a request to Google and perform an email action based on that, not save to the db) and I'm not using any of the form_helper functions. I have little doubt when I get around to validation that's going to be custom as well. To me this defeats the purpose of trying to reuse the Create Project form code, because I'm _not_ reusing most of it, I'm writing a parallel version of it.

Second, conceptually, I think this page is far closer to the pattern for contacting a project or a project leader being able to contact their volunteers. Take this Contact Project component:
Screen Shot 2019-08-22 at 11 25 17 AM

Take that out of a modal and you have the Contact Us form - fill out some formfields and an email is sent on your behalf. I think this should be the pattern I follow.

Admittedly, some of my thinking is that I originally patterned this PR on the contact volunteer/contact project model and it worked and was pretty lightweight in code terms and is no longer either of those things, but I don't think that invalidates my position.

So, long story short, I propose reverting to the version of this PR which was closely patterned on the contact volunteer/contact project email pattern rather than the create project pattern because the purpose is to generate an email and send it on behalf of the user, just like those components. What do you think?
